src/interface.jl

```diff
@@ -1,5 +1,30 @@
 ### AbstractSciMLOperator Interface
 
+"""
+    Set of SciML operator traits
+"""
+Base.@kwdef struct SciMLOperatorTraits{S,O}
+    # Base
+    size::S = nothing
+    #eltype::T
+
+    # LinearAlgebra
+    opnorm::O = nothing
+    isreal::Bool = true
+    issymmetric::Bool = false
+    ishermitian::Bool = false
+
+    # SciML
+    isconstant::Bool = false
+    islinear::Bool = false
+    issquare::Bool = false
+    iszero::Bool = false
+
+    has_adjoint = false
+    has_ldiv = false
+    has_ldiv! = false
+end
```

Comment:
Does this actually resolve at compile time in context? This form seems like it would be relying too much on constant propagation heuristics and would break easily. Also, it's a non-standard way to do traits. I think we should stick to the standard here.

Reply:
TBH i have no idea. im just playing around with different styles to see what sticks. good you caught me before i got in too deep.

Reply:
this isn't relevant to this PR anyway which is about operator composition.

Reply:
im thinking more about traits here: https://github.com/SciML/SciMLOperators.jl/pull/16